lib/entrypoint.sh

```diff
@@ -40,49 +60,90 @@ tag() {
     if [ -n "$INPUT_TAG" ]; then git tag $INPUT_TAG; fi
 }
 
-# This is needed to make the check work for untracked files
-echo "Staging files..."
-add
-remove
-
-echo "Checking for uncommitted changes in the git working tree..."
-# This section only runs if there have been file changes
-if ! git diff --cached --quiet --exit-code; then
-    git_setup
-
-    git fetch
-
-    # Switch branch (create a new one if it doesn't exist)
-    echo "Switching/creating branch..."
-    git checkout "$INPUT_REF" 2>/dev/null || git checkout -b "$INPUT_REF"
+main() {
+    # This is needed to make the check work for untracked files
+    echo "Staging files..."
+    add
+    remove
+    
+    echo "Checking for uncommitted changes in the git working tree..."
+    # This section only runs if there have been file changes
+    if ! git diff --cached --quiet --exit-code; then
+        git_setup
+    
+        git fetch
+    
+        # Switch branch (create a new one if it doesn't exist)
+        echo "Switching/creating branch..."
+        git checkout "$INPUT_REF" 2>/dev/null || git checkout -b "$INPUT_REF"
+    
+        echo "Pulling from remote..."
+        git fetch && git pull
+        git pull --tags
+    
+        echo "Resetting files..."
+        git reset
+    
+        echo "Adding files..."
+        add
+    
+        echo "Removing files..."
+        remove
+    
+        echo "Creating commit..."
+        commit
+    
+        echo "Tagging commit..."
+        tag
+    
+        echo "Pushing commits to repo..."
+        git push --set-upstream origin "$INPUT_REF"
+    
+        echo "Pushing tags to repo..."
+        git push --set-upstream origin "$INPUT_REF" --force --tags
+    
+        echo "Task completed."
+    else
+        echo "Working tree clean. Nothing to commit."
+    fi
+}
 
-    echo "Pulling from remote..."
-    git fetch && git pull
+echo "::endgroup::"
 
-    echo "Resetting files..."
-    git reset
+###############################################################################
 
-    echo "Adding files..."
-    add
+echo "::group::GITCHANGELOG"
+DEBUG_GITCHANGELOG=true gitchangelog
+echo "::endgroup::"
 
-    echo "Removing files..."
-    remove
+###############################################################################
 
-    echo "Creating commit..."
-    commit
+echo "::group::CHANGELOG"
+INPUT_ADD='CHANGELOG.md'
+INPUT_MESSAGE='automatic changelog increase'
+echo $INPUT_MESSAGE
+main
+echo "::endgroup::"
 
-    echo "Tagging commit..."
-    tag
+###############################################################################
 
-    echo "Pushing commits to repo..."
-    git push --set-upstream origin "$INPUT_REF"
+echo "::group::VERSION"
+INPUT_TAG=$(source .version && echo $VERSION)
+INPUT_ADD='.version'
```

Comment:
Can we extract this into it's own variable up top? I think in the future it could be good to have that be a parameter that the user can pass in

Reply:
As in
```suggestion
if [[ -n "$INPUT_ADD" ]]
then
    INPUT_ADD='.version'
fi
```
but at the top?
I can't say I understand why we'd want to pass the version in as a parameter (since one of the point of this is to increment versions), but I'm also not picky about this so if you like I can throw it in.

Reply:
This would just allow a more flexible system in the future. If we decide to add in custom file names, half the process is already done

Reply:
Added!